packages/plugin-core/src/commands/GoToSiblingCommand.ts

```diff
@@ -33,65 +34,207 @@ export class GoToSiblingCommand extends BasicCommand<
-          .map((id) => engine.notes[id])
-          .filter((ent) => _.isUndefined(ent.stub));
-      }
+      sibling =
+        currNoteIdx !== 0
+          ? sortedJournalNotes[currNoteIdx - 1]
+          : _.last(sortedJournalNotes)!;
+    }
+    return { sibling, msg: null };
+  }
+
+  private getSiblingsForJournalNote = (
+    notes: NotePropsByIdDict,
+    currNote: NoteProps
+  ): NoteProps[] => {
+    const month = notes[currNote.parent!];
+    const year = notes[month.parent!];
+    const parent = notes[year.parent!];
+    // Get all descendants of the parent. This includes stub notes
+    const allDescendants = this.getDescendantsRecursively(notes, parent);
+    // Filter out stub notes
+    return allDescendants.filter((note) => !note.stub);
+  };
+
+  private getDescendantsRecursively = (
+    notes: NotePropsByIdDict,
+    parent: NoteProps
+  ): NoteProps[] => {
+    // Get child notes
+    const children = parent.children.map((id) => notes[id]);
+    // Get all descendant notes of the given note, including stub notes
+    const descendants = [...children];
+    // Recursively get descendants
+    for (const child of children) {
+      const grandChild = this.getDescendantsRecursively(notes, child);
+      descendants.push(...grandChild);
     }
+    return descendants;
+  };
 
-    if (respNodes.length <= 1) {
-      window.showInformationMessage(
-        "One is the loneliest number. This node has no siblings :( "
+  private getSibling(
+    workspace: DWorkspaceV2,
+    note: NoteProps,
+    direction: Direction,
+    ctx: string
+  ): { sibling: NoteProps | null; msg: CommandOutput["msg"] | null } {
+    // Get sibling notes
+    const siblingNotes = this.getSiblings(workspace.engine.notes, note);
+    // Check if there is any sibling notes
+    if (siblingNotes.length <= 1) {
+      VSCodeUtils.showMessage(
+        MessageSeverity.INFO,
+        "There is no sibling note in this workspace",
+        {}
       );
-      return {
-        msg: "no_siblings" as const,
-      };
+      return { sibling: null, msg: "no_siblings" };
+    }
+    // Sort the sibling notes
+    const sortedSiblingNotes = this.sortNotes(siblingNotes);
+    // Get the index of the active note in the sorted notes
+    const idx = _.findIndex(sortedSiblingNotes, { id: note.id });
+    // Deal with the unexpected error
+    if (idx < 0) {
+      throw new Error(`${ctx}: ${UNKNOWN_ERROR_MSG}`);
     }
+    // Get sibling based on the direction
+    let sibling: NoteProps;
+    if (direction === "next") {
+      sibling =
+        idx !== siblingNotes.length - 1
+          ? sortedSiblingNotes[idx + 1]
+          : sortedSiblingNotes[0];
+    } else {
+      sibling =
+        idx !== 0 ? sortedSiblingNotes[idx - 1] : _.last(sortedSiblingNotes)!;
+    }
+    return { sibling, msg: null };
+  }
 
+  private getSiblings(
+    notes: NotePropsByIdDict,
+    currNote: NoteProps
+  ): NoteProps[] {
+    if (currNote.fname === "root") {
```

Comment:
another way of doing this without an assertion is checking if `currNote.parent === null`. you don't have to use `currNote.parent!` in the else statement in that case